### DISCUSSION TITLE
Cadastro de produtos e vendas de produtos.

### DISCUSSION THREAD

**Discussion by ameize** [RamonSilva20/mapos#1589] [Ideas]
Quando cadastramos um produto ele pede o valor da compra, se compramos o m² nós sabemos o valor do m² porem alguns produtos são vendidos fracionados, e isso acaba alterando o resultado financeiro.

Sugiro que continuem pedindo o valor de custo do m² e o valor de venda, mas na hora de vender o produto por m² tenha o campo "ALTURA" E "LARGURA" assim quando for preenchido o sistema deve fazer a multiplicação simples Altura x Largura x Valor de venda do m², assim no relatório financeiro você tem exato quantos m² foi vendido, quanto te custou e quando ganhou.

O sistema é ótimo, parabéns pelo trabalho!.

**Comment by SuperPorte** [comment]
Complexo isso hein???
Como seria o estoque dessa altura e largura?
Tipo, dá entrada de 200mA + 750mL e ir vendendo ambos juntos, porém separados?
Se vende-se 1mA com 750mL como faria?

Deu nó no meu cérebro hehehe

**Reply by ameize** [reply]
Boa tarde Bernarcom, tudo bem?

Na verdade não é tão complexo, provavelmente eu me expressei mal, vou tentar simplificar citando um exemplo de uma empresa fictícia de comunicação visual / gráfica.

**Cadastro do produto**
```
A empresa Bernar Comunicação compra adesivo em bobina, cada bonina custa R$: 300reais e tem 50m².
O Custo da Bernar Comunicação é de R$: 6,00 por m² e ele definiu o valor de venda a R$: 20,00 m².
```

**Venda do produto**
```
Quando um colaborador da Bernar Comunicação for vender este produto, havera o nome do produto, a quantidade e 3 campos adicionais,
Altura / Largura / Valor do m² quando o colaborador preencher a altura e a largura o sistema faz uma multiplicação simples dos campos.

Altura x Largura x valor do m2 (no caso da Bernar Comunicação R$:20,00)
```

**Exemplo de venda**
```
Um cliente precisa de um adesivo 2mt x 1,5mt o colaborador vai preencher... 
(Altura - 2mt) - (Largura 1.5mt) - (Valor do m² - R$:20,00) após a multiplicação o sistema já gera o valor de venda, que nesse caso seria R$:60,00

O sistema sabe que R$:6,00 é o valor de custo do m² e que R$:20,00 é o valor de venda, assim com a mesma multiplicação simples consegue saber custo, venda e lucro.
```

**Exemplo de estoque**

`Ainda seguindo o exemplo de venda acima, multiplicando altura x largura (2mt x 1,5mt) o sistema sabe que foi utilizado 3mt do produto, ou seja, se ele comprou uma bobina de 50metros, após a venda, resta no estoque um total de 47m²`

**Comment by SuperPorte** [comment]
A venda eu entendi. Estou pensando no estoque.Já tentou fazer no Excel? Deu certo? Manda a formula.Em 27 de dez de 2021 18:05, ameize ***@***.***> escreveu:

Boa tarde Bernarcom, tudo bem?

Na verdade não é tão complexo, provavelmente eu me expressei mal, vou tentar simplificar citando um exemplo de uma empresa fictícia de comunicação visual / gráfica.

Cadastro do produto

A empresa Bernar Comunicação compra adesivo em bobina, cada bonina custa R$: 300reais e tem 50m².

O Custo da Bernar Comunicação é de R$: 6,00 por m² e ele definiu o valor de venda a R$: 20,00 m².



Venda do produto

Quando um colaborador da Bernar Comunicação for vender este produto, havera o nome do produto, a quantidade e 3 campos adicionais,

Altura / Largura / Valor do m² quando o colaborador preencher a altura e a largura o sistema faz uma multiplicação simples dos campos.



Altura x Largura x valor do m2 (no caso da Bernar Comunicação R$:20,00)



Exemplo de venda

Um cliente precisa de um adesivo 2mt x 1,5mt o colaborador vai preencher... 

(Altura - 2mt) - (Largura 1.5mt) - (Valor do m² - R$:20,00) após a multiplicação o sistema já gera o valor de venda, que nesse caso seria R$:60,00



O sistema sabe que R$:6,00 é o valor de custo do m² e que R$:20,00 é o valor de venda, assim com a mesma multiplicação simples consegue saber custo, venda e lucro.



Exemplo de estoque

Ainda seguindo o exemplo de venda acima, multiplicando altura x largura (2mt x 1,5mt) o sistema sabe que foi utilizado 3mt do produto, ou seja, se ele comprou uma bobina de 50metros, após a venda, resta no estoque um total de 47m²

**Reply by ameize** [reply]
Boa noite. Basicamente, após a soma da LARGURA X ALTURA na venda o software vai identificar quantos m² foram vendidos, é só subtrair da quantidade de m² que o propietario colocou como estoque.

**O Código no Excel seria...**
```
LINHA A1 - ALTURA / LINHA A2 - LARGURA / LINHA A3 - VALOR DO M² / LINHA A4 - ESTOQUE ATUAL
Para calcular estoque utiliza =SOMA(A1-A3)
Para venda utiliza =SOMA(A1*A2*A3)
```

O Código para o sistema mapos seria algo em torno disso...

```
                                    <div class="span2">
                                        <label for="">Altura</label>
                                        <input type="text" placeholder="Altura" id="$numero1" name="numero1" class="span12 money" />
                                    </div>
                                    <div class="span2">
                                        <label for="">Largura</label>
                                        <input type="text" placeholder="Largura" id="$numero2" name="numero2" class="span12 money" />
                                    </div>
                                    <div class="span2">
                                        <label for="">Valor do m²</label>
                                        <input type="text" placeholder="Valor do m²" id="$numero3" name="numero3" class="span12 money" />
                                    </div>
                                    <div class="span2">
                                        <label for="">Quantidade</label>
                                        <input type="text" placeholder="Quantidade" id="$quantidade" name="quantidade" class="span12" />
                                    </div>
                                    <div class="span2">
                                        <label for="">Valor total</label>
                                        <?php echo $total = $numero1 * $numero2 * $numero3 * $quantidade; echo $total; ?>
                                    </div>
```

Sou entusiasta, não fiz cursos ou algo do tipo! o código EXCEL com certeza está correto, oq sugeri para o MAPOS pode contar um erro ou outro, mas acredito estar corretio.

**Comment by ameize** [comment]
Ao adicionar o código que passei acima o visual torna-se idêntico ao print abaixo.
https://imgur.com/a/gr7kxCv

Porém, não consigo tornar usual a ponto do sistema calcular pra mim.

**Reply by jooilson** [reply]
Boa noite!
Acho entendi. Ao invés do sistema calcular de maneira automática, o usuário irá inserir os valores.
Correto?

**Reply by ameize** [reply]
Boa noite Jooilson, tudo bem? Sim, o usuário coloca a altura, largura e valor de venda do m² e o software multiplica isso e gera o valor.

**Reply by guskid** [reply]
Utilizo o mapos aqui pra gráfica que tenho tb.. 
Faço o Calculo manualmente e crio um novo produto sempre que vendo adesivo ou lona. 
Se vocês conseguissem criar esse sitema para m² seria top pacas...
